packages/contracts/test/core/dao/dao.ts

```diff
@@ -1190,98 +1217,176 @@ describe('DAO', function () {
-      await expect(
-        dao
-          .connect(signers[2])
-          .setSignatureValidator(ethers.Wallet.createRandom().address)
-      )
+    it('allows caller-specific signature validation conditions', async () => {
+      // Try to call with caller but caller has no permission
+      expect(await dao.connect(caller).isValidSignature(hash, signature))
         .to.be.revertedWithCustomError(dao, 'Unauthorized')
-        .withArgs(
-          dao.address,
-          signers[2].address,
-          PERMISSION_IDS.SET_SIGNATURE_VALIDATOR_PERMISSION_ID
-        );
-    });
+        .withArgs();
+
+      // Deploy a mock condition
+      const mockCondition = await new PermissionConditionMock__factory(
+        caller
+      ).deploy();
 
-    it('should set validator and emits event', async () => {
-      const validatorAddress = ethers.Wallet.createRandom().address;
-      const tx = await dao.setSignatureValidator(validatorAddress);
+      // Grant the permission to validate signatures to the caller
+      await dao.grantWithCondition(
+        dao.address,
+        caller.address,
+        PERMISSION_IDS.VALIDATE_SIGNATURE_PERMISSION_ID,
+        mockCondition.address
+      );
 
-      expect(await dao.signatureValidator()).to.be.eq(validatorAddress);
+      // The caller can validate signatures now.
+      expect(await dao.connect(caller).isValidSignature(hash, signature)).to.not
+        .be.reverted;
 
-      await expect(tx)
-        .to.emit(dao, EVENTS.SignatureValidatorSet)
-        .withArgs(validatorAddress);
-    });
+      // Check that the mock condition will answer true.
+      expect(await mockCondition.answer()).to.be.true;
+
+      // Check that the signature is invalid in this case.
+      expect(
+        await dao.connect(caller).isValidSignature(hash, signature)
+      ).to.equal(VALID_ERC1271_SIGNATURE);
+
+      // Set the mock condition to answer false.
+      await mockCondition.setAnswer(false);
 
-    it('should call the signature validator', async () => {
-      const ERC1271MockFactory = await smock.mock('ERC1271Mock');
-      const erc1271Mock = await ERC1271MockFactory.deploy();
+      // Check that the mock condition will answer false.
+      expect(await mockCondition.answer()).to.be.false;
 
-      await dao.setSignatureValidator(erc1271Mock.address);
-      await dao.isValidSignature(ethers.utils.keccak256('0x00'), '0x00');
-      expect(erc1271Mock.isValidSignature).has.been.callCount(1);
+      // Check that the signature is invalid in this case.
+      expect(
+        await dao.connect(caller).isValidSignature(hash, signature)
+      ).to.equal(INVALID_ERC1271_SIGNATURE);
     });
 
-    it('should return the validators response', async () => {
-      const ERC1271MockFactory = new ERC1271Mock__factory(signers[0]);
-      const erc1271Mock = await ERC1271MockFactory.deploy();
+    it('allows generic signature validation by granting to ANY_ADDR', async () => {
```

Comment:
missing check if `isValidSignature` returns `VALID_ERC1271_SIGNATURE` when it is valid

Reply:
Fixed in 89817bd419c478c7130139366f6b24c53c7ca9a7